runtime/tests/org.eclipse.e4.core.tests/src/org/eclipse/e4/core/internal/tests/di/extensions/InjectionOSGiTest.java

```diff
@@ -109,6 +110,13 @@ public void setUp() throws Exception {
 
 	@Test
 	public void ensureJavaxIsNotAvailable() {
+		// This entire test-plugin is not really useful if javax.inject and
+		// jakarta.inject is available and this test-case fails then.
+		// However due to the way I-build tests are set up (i.e. by using a built
+		// Eclipse installation) exactly that's the case.
+		// -> Disable this for I-build tests
+		Assume.assumeFalse("org.eclipse.ant.core.antRunner".equals(System.getProperty("eclipse.application")));
```

Comment:
How is that any better then my solution? It has the same downside that accidentally this condition may change in future on the various build types or when running the test locally.

Reply:
Because with your suggestion in #1364 that case cannot fail at all and therefore becomes useless because it cannot indicate that `javax.inject` is present although it is expected to be absent:
https://github.com/eclipse-platform/eclipse.platform/pull/1364#issuecomment-2107758417

This solution only disables that case for I-builds since it cannot really be changed that `javax.inject` is present (at least I failed to implement the dynamic removal).

> It has the same downside that accidentally this condition may change in future on the various build types or when running the test locally.

Yes, but that's intended. If in the future I-builds use another application that condition has to be adapted. But I think that's the smaller evil. And in local builds the test should still fail if javax.inject is present to make developers aware that the current setup is not as it should be.